ivy/func_wrapper.py

```diff
@@ -812,15 +812,25 @@ def _handle_device_shifting(*args, **kwargs):
         -------
             The return of the function.
         """
+        dev = None
+        if "device" in kwargs and kwargs["device"] is not None:
+            dev = ivy.as_native_dev(kwargs["device"])
         if ivy.soft_device_mode:
-            return ivy.handle_soft_device_variable(*args, fn=fn, **kwargs)
+            return ivy.handle_soft_device_variable(*args, fn=fn, dst_dev=dev, **kwargs)
```

Comment:
let's rename `dst_dev` to `device_shifting_dev` for code clarity. Same applies to all other occurrences of `dst_dev`